pkg/alertmanager/distributor_test.go

```diff
@@ -330,6 +336,7 @@ func prepare(t *testing.T, numAM, numHappyAM, replicationFactor int, responseBod
 
 	return d, ams, func() {
 		require.NoError(t, services.StopAndAwaitTerminated(context.Background(), d))
+		require.NoError(t, services.StopAndAwaitTerminated(context.Background(), amRing))
```

Comment:
Drive-by fix: enabling the leak detector for `TestMultitenantAlertmanager_loadAndSyncConfigs` would fail due to unrelated goroutines leaked by this test. , This test only happens to fall earlier in the test ordering.